internal/monitors/haproxy/config.go

```diff
@@ -23,3 +42,27 @@ type Config struct {
 	// If empty then metrics for all proxies will be reported.
 	Proxies []string `yaml:"proxies"`
 }
+
+// Validate the config
+func (c *Config) Validate() error {
```

Comment:
This validation is too constraining.  There are various edge-cases where it is useful to be able to mix and match discovery rules and configs in ways that we cannot foresee, so I am more inclined towards letting the user do whatever they want.  As long as it is clearly documented how the various config options relate to each other (I think it is), then it should be fine.

Get rid of this method.

Reply:
ok